kouncil-frontend/apps/kouncil/src/app/schemas/form/form/schema-form.component.ts

```diff
@@ -0,0 +1,182 @@
+@Component({
+  selector: 'app-schema-form',
+  template: `
+    <form *ngIf="model" (ngSubmit)="saveSchema()" class="form schema-form"
+          #schemaForm="ngForm">
+
+      <div class="schema-base-info">
+        <div>
+          <div class="label">Topic</div>
+          <mat-form-field [appearance]="'outline'">
+            <mat-select name="topicName" [(ngModel)]="model.topicName"
+                        [disabled]="isDisabled([ViewMode.VIEW, ViewMode.EDIT])"
+                        [required]="isRequired([ViewMode.CREATE])"
+                        #topicName="ngModel">
+              <mat-option *ngFor="let topic of topics" [value]="topic.value">
+                {{ topic.label }}
+              </mat-option>
+            </mat-select>
+          </mat-form-field>
+        </div>
+
+        <div>
+          <div class="label">Subject type</div>
+          <mat-form-field [appearance]="'outline'">
+            <mat-select name="subjectType" [(ngModel)]="model.subjectType"
+                        [disabled]="isDisabled([ViewMode.VIEW, ViewMode.EDIT])"
+                        [required]="isRequired([ViewMode.CREATE])">
+              <mat-option *ngFor="let subjectType of subjectTypes" [value]="subjectType">
+                {{ subjectType }}
+              </mat-option>
+            </mat-select>
+          </mat-form-field>
+        </div>
+        <div *ngIf="isVisible([ViewMode.VIEW])">
+          <div class="label">Versions</div>
+          <mat-form-field [appearance]="'outline'">
+            <mat-select name="version" [(ngModel)]="model.version" (selectionChange)="changeSchemaVersion($event)">
+              <mat-option *ngFor="let version of model.versionsNo" [value]="version">
+                {{ version }}
+              </mat-option>
+            </mat-select>
+          </mat-form-field>
+        </div>
+        <div>
+          <div class="label">Message format</div>
+          <mat-form-field [appearance]="'outline'">
+            <mat-select name="messageFormat" [(ngModel)]="model.messageFormat"
+                        [disabled]="isDisabled([ViewMode.VIEW, ViewMode.EDIT])"
+                        [required]="isRequired([ViewMode.CREATE])">
+              <mat-option *ngFor="let messageFormat of messageFormats" [value]="messageFormat">
+                {{ messageFormat }}
+              </mat-option>
+            </mat-select>
+          </mat-form-field>
+        </div>
+        <div>
+          <div class="label">Compatibility</div>
+          <mat-form-field [appearance]="'outline'" style="width: 250px">
+            <mat-select name="compatibility" [(ngModel)]="model.compatibility"
+                        [disabled]="isDisabled([ViewMode.VIEW])">
+              <mat-option *ngFor="let compatibility of compatibilities" [value]="compatibility">
+                {{ compatibility }}
+              </mat-option>
+            </mat-select>
+            <button *ngIf="model.compatibility && !isDisabled([ViewMode.VIEW])"
```

Comment:
Why only Compatibility dropdown has this clear button?

Reply:
Because this is the only field that can be empty